pilot/cmd/pilot-agent/status/server.go

```diff
@@ -502,8 +514,51 @@ func negotiateMetricsFormat(contentType string) expfmt.Format {
 	return expfmt.FmtText
 }
 
-func processMetrics(metrics []byte) []byte {
-	return bytes.ReplaceAll(metrics, []byte("\n\n"), []byte("\n"))
+// metricReader used to replace "\n\n" in stream with '\n'.
```

Comment:
I think this is missing an explanation about *why* we do this?

Reply:
@howardjohn The code comments have been completed